tests/workload/transactions.go

```diff
@@ -0,0 +1,125 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package workload
+
+import (
+	"context"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/ava-labs/hypersdk/chain"
+	"github.com/ava-labs/hypersdk/rpc"
+	"github.com/ava-labs/hypersdk/utils"
+)
+
+// TxWorkloadFactory prescribes an exact interface for generating transactions to test on a given environment
+// and a sized sequence of transactions to test on a given environment and reach a particular state
+type TxWorkloadFactory interface {
+	// Generate a new TxWorkloadIterator that generates a single sequence of transactions
+	// and corresponding assertions.
+	NewBasicTxWorkload(uri string) (TxWorkloadIterator, error)
+	// Generates a new TxWorkloadIterator that generates a sequence of transactions of the given size.
+	NewSizedTxWorkload(uri string, size int) (TxWorkloadIterator, error)
+}
+
+// TxWorkloadIterator provides an interface for generating a sequence of transactions and corresponding assertions.
```

Comment:
Adding

```golang
// This pattern is intended to allow the workload to define how many transactions it should support. For example,
// a CFMM application may define a set of workloads that define each sequence of actions that should be tested
// against different network configurations such as:
// 1. Create Liquidity Pool
// 2. Add Liquidity
// 3. Swap
```

There may be a better way to do this, but the intent was to support two use cases. For the current sync tests, the HyperSDK defined test needs a way to generate a sequence of transactions of arbitrary length, so that it can trigger different phases of bootstrapping/state sync.

In addition, VM developers should be able to define their own sequences of transactions to run against an arbitrary environment where the VM defines how many transactions it should issue.

The transactions must be generated on demand instead of returning a slice all at once to avoid transactions expiring before they can be issued to the test environment.